src/main/java/no/finn/unleash/DefaultUnleash.java

```diff
@@ -95,6 +100,51 @@ public boolean isEnabled(final String toggleName, final UnleashContext context ,
         return enabled;
     }
 
+    @Override
+    public Variant getVariant(String toggleName, UnleashContext context) {
+        return getVariant(toggleName, context, "");
+    }
+
+    @Override
+    public Variant getVariant(String toggleName, UnleashContext context, String defaultPayload) {
+        final  FeatureToggle featureToggle = toggleRepository.getToggle(toggleName);
+        final boolean isEnabled = isEnabled(toggleName, context, false, featureToggle);
+
+        if(isEnabled) {
+            return selectVariant(featureToggle, context)
+                .map(variantDefinition -> new Variant(variantDefinition.getName(), variantDefinition.getPayload(), isEnabled))
+                .orElseGet(() -> new Variant("default", defaultPayload, isEnabled));
+        } else {
+            return new Variant("default", defaultPayload, isEnabled);
+        }
+    }
+
+    private Optional<VariantDefinition> selectVariant(final FeatureToggle featureToggle, UnleashContext context) {
+        if(featureToggle.getVariants() == null || featureToggle.getVariants().isEmpty()){
+            return Optional.empty();
+        }
+
+        final int sum = featureToggle.getVariants().stream().mapToInt(VariantDefinition::getWeight).sum();
+        final int score = 1 + StrategyUtils.getNormalizedNumber(
+            context.getUserId()
+                .orElse(context.getSessionId()
+                    .orElse(context.getRemoteAddress()
+                        .orElse(""))),
+            featureToggle.getName(),
+            sum);
+
+        int num = 0;
+        for (VariantDefinition definition: featureToggle.getVariants()){
+            if(score + num >= definition.getWeight()){
+                return Optional.of(definition);
+            }
+            num += definition.getWeight();
+        }
+
+        //Should not happen
```

Comment:
What should not happen?

Reply:
Edge case when we are done with the loop. If the result of the modulus operator is higher than max. This is now treated as return empty as we talked about in the discussion this morning.